libbpfgo.go

```diff
@@ -564,6 +564,14 @@ func (b *BPFMap) Type() MapType {
 	return MapType(C.bpf_map__type(b.bpfMap))
 }
 
+func (b *BPFMap) MapReuseFd(fd int) error {
+	errC := C.bpf_map__reuse_fd(b.bpfMap, C.int(fd))
+	if errC != 0 {
+		return fmt.Errorf("could not reuse bpf map: %w", syscall.Errno(-errC))
+	}
+	return nil
```

Comment:
Before return, we need to update `b.fd` to the `new_fd` via `C.bpf_map__fd(b.bpfMap)`.